### PR TITLE
feat: created protected route and useDocumentTitle hook

### PR DESCRIPTION
**Pull request by Developers-codz**
## Files To Check
`src/component/NonAuthenticatedRoute.jsx`
`src/component/ProtectedRoute.jsx`
`src/context/auth-context.js`
`src/hooks/useDocumentTitle.js`
`src/pages/login/Login.jsx`
`src/pages/profile/Profile.jsx`
`src/pages/signup/Signup.jsx`
`src/reducer/auth-reducer.js`

## Files not to check
Please do not check other files.

thank you :)

## [Live Link](https://deploy-preview-7--distracted-payne-21e589.netlify.app/)

### CODE REVIEW

File: src/context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        e.target.value === "user"
+          ? JSON.stringify(formData)
+          : {
+              email: "kajalkumari@gmail.com",
+              password: "kajalkumari",
+            }
+      );
+      //saving the token in localstorage
+      localStorage.setItem("token", response.data.encodedToken);
+      console.log(response.data.foundUser);
```
Comment (sainath7878) [inline src/context/auth-context.js:24]: remove console.log

File: src/pages/login/Login.jsx [hunk 2]
```diff
@@ -39,23 +53,32 @@ const Login = () => {
...
             <button
               className="login-btn reset"
               type="submit"
               value="user"
-              onClick={(e) => loginHandler(e, formData)}
+              onClick={(e) => clickHandler(e)}
             >
               Login{" "}
             </button>
+            <div style={{ color: "red" }}>{msg}</div>
```
Comment (sainath7878) [inline src/pages/login/Login.jsx:81]: avoid inline styling

**Review by sainath7878 — APPROVED** [review]
Your code looks neat and clean. Good work!

File: src/reducer/auth-reducer.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    case "signup": {
+      const { firstName, lastName, _id } = action.payload;
+      const token = action.token;
+      return {
+        ...state,
+        id: _id,
+        firstName: firstName,
+        lastName: lastName,
+        isAuthTokenPresent: true,
+        authToken: token,
```
Comment (ApurvaSawant11) [inline src/reducer/auth-reducer.js:35]: You are already assigning the token to `authToken`. So, I think you don't need an extra indicator to check if the token is present or not.

**Review by ApurvaSawant11 — APPROVED** [review]
The code looks good. Just add consistent spacing in the files.

File: src/pages/login/Login.jsx (hunk 2 above)
Comment (Developers-codz) [reply src/pages/login/Login.jsx:81]: okay

File: src/context/auth-context.js (hunk 1 above)
Comment (Developers-codz) [reply src/context/auth-context.js:24]: Will remove it after implementing cart wishlist functionality

File: src/reducer/auth-reducer.js (hunk 3 above)
Comment (Developers-codz) [reply src/reducer/auth-reducer.js:35]: okay will look into that thank you :)

**Pull request merged**